cloudinit/cmd/main.py

```diff
@@ -1122,8 +1144,69 @@ def main(sysv_args=None):
 
             status_parser(parser_status)
             parser_status.set_defaults(action=("status", handle_status_args))
+    else:
+        parser.error("a subcommand is required")
 
     args = parser.parse_args(args=sysv_args)
+    setattr(args, "skip_log_setup", False)
+    if not args.all_stages:
+        return sub_main(args)
+    LOG.info("Running cloud-init in single process mode.")
```

Comment:
Worth a separate function here for run_single_process_mode. The main func is already 200 lines long before we added the single proc logic.

Reply:
Added in latest commit